osbuild/objectstore.py

```diff
@@ -137,13 +147,9 @@ def commit(self, tree: TreeObject, object_id: str) -> str:
         # the tree is stored in the objects directory using its content
         # hash as its name, ideally a given object_id (i.e., given config)
         # will always produce the same content hash, but that is not
-        # guaranteed
-        output_tree = f"{self.objects}/{treesum_hash}"
-
-        # if a tree with the same treesum already exist, use that
-        with suppress_oserror(errno.ENOTEMPTY, errno.EEXIST):
-            os.rename(tree.path, output_tree)
-            tree.path = output_tree
+        # guaranteed. If a tree with the same treesum already exist, it
+        # will be overwritten
```

Comment:
Hm, this isn't right, is it? If a tree with the same treesum exists, the current one will be discarded and the existing one will be used instead.

Reply:
You are of course right. This is more than embarrassing. Lets pretend it never happened, ok?

Reply:
Deal!